examples/40_advanced/example_inspect_predictions.py

```diff
@@ -0,0 +1,109 @@
+# -*- encoding: utf-8 -*-
+"""
+=================
+Model Explanation
+=================
+
+The following example shows how to fit a simple classification model with
+*auto-sklearn* and use the inspect `module <https://scikit-learn.org/stable/inspection.html>`_ from
+scikit-learn to understand what affects the predictions.
+"""
+import sklearn.datasets
+from sklearn.inspection import plot_partial_dependence, permutation_importance
+import matplotlib.pyplot as plt
+import autosklearn.classification
+
+
+################################
+# Load Data and Build a Model
+# ==============================
+#
+# We use the Run or walk dataset as given by OpenML. The goal is to predict whether a person is
+# running or walking based on accelerometer and gyroscope data collected by a phone. For more
+# information see `here <https://www.openml.org/d/40922>`_.
+
+dataset = sklearn.datasets.fetch_openml(data_id=40922)
+
+# Note: To speed up the example, we subsample the dataset
+dataset.data = dataset.data.sample(n=5000, random_state=1, axis="index")
+dataset.target = dataset.target[dataset.data.index]
+X_train, X_test, y_train, y_test = \
+    sklearn.model_selection.train_test_split(dataset.data, dataset.target, test_size=0.3,
+                                             random_state=1)
+
+automl = autosklearn.classification.AutoSklearnClassifier(
+    time_left_for_this_task=360,
+    per_run_time_limit=30,
+    tmp_folder='/tmp/autosklearn_classification_example_tmp',
+    output_folder='/tmp/autosklearn_classification_example_out',
+    n_jobs=1,
+)
+automl.fit(X_train, y_train, dataset_name='Run_or_walk_information')
+
+s = automl.score(X_train, y_train)
+print(f"Train score {s}")
+s = automl.score(X_test, y_test)
+print(f"Test score {s}")
+
+#########################################
+# Compute permutation importance - part 1
+# =======================================
+#
+# Since auto-sklearn implements the scikit-learn interface, it can be used with the scikit-learn's
+# inspection module. So, now we first look at the `permutation importance
+# <https://christophm.github.io/interpretable-ml-book/feature-importance.html>`_, which defines the
+# decrease in a model score when a given feature is randomly permutated. So, the higher the score,
+# the more does the model's predictions depend on this feature.
+#
+# **Note:** There are some pitfalls in interpreting these numbers, which can be found
+# `here <https://scikit-learn.org/stable/modules/permutation_importance.html>`_.
+
+r = permutation_importance(automl, X_test, y_test,
+                           n_repeats=10,
+                           random_state=0)
+
+sort_idx = r.importances_mean.argsort()[::-1]
+plt.boxplot(r.importances[sort_idx].T, labels=[dataset.feature_names[i] for i in sort_idx])
+plt.xticks(rotation=90)
```

Comment:
I think we should use something like [tight_layout](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.tight_layout.html) as at least for me the names are cut? Or probably rotate it a bit.

Reply:
Agree. I added tight_layout() to all plots s.t. they hopefully look better now

Reply:
Agree